client-linuxapp/src/main.rs

```diff
@@ -38,6 +43,53 @@ const RV_DEFAULT_DELAY_SEC: f32 = 120.0;
 const RV_DEFAULT_DELAY_OFFSET: f32 = 30.0;
 const RV_USER_DEFINED_DELAY_OFFSET: f32 = 0.25;
 
+// Encapsulates errors caused during TO1/TO2
+#[derive(Debug)]
+struct ErrorResult {
+    e_code: ErrorCode,
+    e_string: String,
```

Comment:
As far as I can find, this is always just a hardcoded string that you type in the code explicitly.
You could use `e_string: &'static str` instead then, that avoids having to create a `String` object, and is slightly neater code throughout.
Normally, when you have a borrowed object in a `struct`, it needs an explicit lifetime that the `struct` itself also has, but `'static` is special, in that it doesn't require that: that indicates that the (in this case string) is statically compiled into the binary, so it has a lifetime that exceeds any function we define.